File: codility/lessons/5 - prefix sums/count div/count_div.cpp
// Source: https://app.codility.com/programmers/lessons/5-prefix_sums/count_div/

int solution(int begin, int end, int divisor)
{
    int dividend {(end - begin) / divisor};

    if (!(begin % divisor) || !(end % divisor)
            || (begin < divisor && (end % divisor + divisor - begin) < divisor))
        dividend++;

    return dividend;
}

File: codility/lessons/5 - prefix sums/count div/count_div.rb
# Source: https://app.codility.com/programmers/lessons/5-prefix_sums/count_div/

def solution(range_begin, range_end, divisor)
    dividend = ((range_end - range_begin) / divisor).truncate

    if range_begin % divisor == 0 or range_end % divisor == 0 \
        or (range_begin < divisor and (range_end % divisor + divisor - range_begin) < divisor)
        dividend += 1
    end

   dividend
end
